### PR TITLE
Update property URL path in UI

### PR DESCRIPTION
**Pull request by edmundito**
## Change description

Because properties are part of of sources, this updates the path of a property from:

```
/model/{modelId}/property/{propertyId}
```

to:
```
/model/{modelId}/source/{sourceId}/property/{propertyId}
```

- Moved the pages and updated link paths
- Added `sourceId` to property recordPreview so we can generate the link to the property from the sample record card correctly
- Rebuild Tabs so that breadcrumbs could manage more nested paths

### Development

- [x] Application changes have been tested appropriately

### Impact

- [ ] Code follows company security practices and guidelines
- [ ] Security impact of change has been considered
- [ ] Performance impact of change has been considered
- [ ] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:

### Code review

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [x] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

**Comment by edmundito** [comment]
Meant to be a Draft because I have one more change left to do, but I accidentally posted it as a final PR. You can start taking a look at what I got so far.

**Comment by edmundito** [comment]
I wrapped up the work on the breadcrumbs and this is ready for review.

File: ui/ui-components/components/Tabs.tsx [hunk 1]
```diff
@@ -3,76 +3,123 @@ import Link from "next/link";
...
-  const pathnameScope = pathnameParts
-    .slice(0, pathnameParts.length - 3)
-    .join("/");
-  const pathnameId = pathnameParts[pathnameParts.length - 2];
+      router.push(
+        `${pathnameScope}/${topic}/${pathnameId}/${tabName}`,
+        `${scope}/${topic}/${id}/${tabName}`
+      );
+    };
+  }, [parts, pathnameParts]);
```
Comment (krishnaglick) [inline ui/ui-components/components/Tabs.tsx:58]: Looking at this makes me thing it should be a set of useMemo's for scope, topic, etc, and then a useCallback for the actual function.

File: ui/ui-components/pages/model/[modelId]/source/[sourceId]/property/[propertyId]/edit.tsx [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
+import PropertyTabs from "../../../../../../../components/tabs/Property";
+import { Models, Actions } from "../../../../../../../utils/apiData";
+import { filtersAreEqual } from "../../../../../../../utils/filtersAreEqual";
+import { makeLocal } from "../../../../../../../utils/makeLocal";
+import { ErrorHandler } from "../../../../../../../utils/errorHandler";
+import { SuccessHandler } from "../../../../../../../utils/successHandler";
+import { PropertiesHandler } from "../../../../../../../utils/propertiesHandler";
+import ModelBadge from "../../../../../../../components/badges/ModelBadge";
+import { ensureMatchingModel } from "../../../../../../../utils/ensureMatchingModel";
+import PropertySampleRecord from "../../../../../../../components/property/PropertySampleRecord";
```
Comment (krishnaglick) [inline ui/ui-components/pages/model/[modelId]/source/[sourceId]/property/[propertyId]/edit.tsx:26]: Relative paths gonna be clutch here

File: ui/ui-components/components/Tabs.tsx (hunk 1 above)
Comment (edmundito) [reply ui/ui-components/components/Tabs.tsx:58]: I thought about using a callback hook but scope, topic, etc is not really used anywhere else in the component and it shouldn't. So it made sense to me to just have it run through useMemo once.

File: core/src/modules/ops/record.ts [hunk 3]
```diff
@@ -26,32 +26,35 @@ import { GrouparooModel } from "../../models/GrouparooModel";
...
-    valueChangedAt: RecordProperty["valueChangedAt"];
-    confirmedAt: RecordProperty["confirmedAt"];
-    stateChangedAt: RecordProperty["stateChangedAt"];
-    startedAt: RecordProperty["startedAt"];
-    createdAt: RecordProperty["createdAt"];
-    updatedAt: RecordProperty["updatedAt"];
-  };
+export interface RecordPropertyValue {
+  id: RecordProperty["id"];
+  sourceId: Property["sourceId"];
```
Comment (evantahler) [inline core/src/modules/ops/record.ts:31]: 👍

**Review by evantahler — APPROVED** [review]
Nice! I like the changes to the logic in the Tab component, and it looks like the only additional data needed from the API was (a consistent) sourceId

**Pull request merged**